src/contexts/Maci.tsx

```diff
@@ -241,10 +241,16 @@ export const MaciProvider: React.FC<MaciProviderProps> = ({ children }: MaciProv
         provider: signer.provider,
       })
         .then((data) => {
-          setPollData(data);
+          if (
+            (pollData && (data.address !== pollData.address || data.isStateAqMerged !== pollData.isStateAqMerged)) ||
```

Comment:
This should be moved to a separate const
```
(data.address !== pollData.address || data.isStateAqMerged !== pollData.isStateAqMerged)
```

Reply:
I made this an independent const function, should I just put inside this MACI context? or I think it could be in utils? (not sure if someone else need it)